com.unity.render-pipelines.core/Runtime/Debugging/DebugUpdater.cs

```diff
@@ -32,20 +37,90 @@ internal static void SetEnabled(bool enabled)
 
         static void EnableRuntime()
         {
+            if (s_Instance != null)
+                return;
+
             var go = new GameObject { name = "[Debug Updater]" };
-            var debugUpdater = go.AddComponent<DebugUpdater>();
+            s_Instance = go.AddComponent<DebugUpdater>();
```

Comment:
It is weird to see the initialization of the SIngleton outside the `instance` property. Can we move it to that property?

Reply:
Hmm.  I don't think I could move it to the property, because it's a component so we need a gameobject to initialize it. This is not a traditional singleton in the sense that it has explicit lifetime (i.e. it will be created _and_ destroyed at will, depending on how `SetEnabled(bool)` has been called), so I think it's more clear to keep the create/destroy here. Moving that logic to the property might give the false impression that it's always available, which is not the case.